pkg/network/ebpf/c/protocols/http2/decoding.h

```diff
@@ -117,29 +129,35 @@ static __always_inline void parse_field_indexed(dynamic_table_index_t *dynamic_i
 
 READ_INTO_BUFFER(path, HTTP2_MAX_PATH_LEN, BLK_SIZE)
 
-// parse_field_literal handling the case when the key is part of the static table and the value is a dynamic string
-// which will be stored in the dynamic table.
+// parse_field_literal parses headers with the header name being in the static
+// table and the value is a literal string to be stored in the dynamic table.
```

Comment:
> // parse_field_literal parses headers with the header name being in the static
// table

That's not entirely true. We are handling a case where the header name is being passed directly as well.
When the index is 0, the header name is passed by value. (See https://httpwg.org/specs/rfc7541.html#detailed.format)